pydantic_ai_slim/pydantic_ai/retries.py

```diff
@@ -179,17 +181,33 @@ def handle_request(self, request: Request) -> Response:
 
         @retry(**self.config)
         def handle_request(req: Request) -> Response:
-            response = self.wrapped.handle_request(req)
+            with self:
```

Comment:
This may not work right if the transport was already entered manually through entering the client... Should we instead add an on-error handler to exit and enter again?